### ISSUE TITLE
does activity that was always in null change least?

### ISSUE THREAD

**Issue by mobeets** [mobeets/nullSpaceControl#149]
or the converse, which might actually be more interesting: does the part that used to be row space change at all, or relax? this might even be the key question of the project, because this is where you'd expect to see relaxation towards minimal/baseline firing, say, or else movement towards a more uniform distribution, etc.

**Comment by mobeets** [comment]
These values are very consistent across days, even across monkeys:

![screen shot 2016-05-06 at 9 59 07 am](https://cloud.githubusercontent.com/assets/1677179/15075215/32616928-1371-11e6-8f1c-3785877445f5.png)

The y-axis is always the median percent of norm that is in the null space. So for each time point, if `a=colnorm(Y*(NB*NB'))`, `b=colnorm(Y*(RB*RB'))`, then the y-value is `median(a/(a+b))`.

So the first bar is the percent of the norm in Blk1 that's in its null space. Similarly for the second bar.
The third and fourth bars are the percent in Blk1's null space, where `YN2 = Y2*(NB2*NB2')`.

So basically:
- bars 1 and 2 tell us that there's more activity in the respective null spaces during Blk2 than for Blk1.
- bar 3 tells us that around 60% of the activity in the Blk2 null space comes from activity that was also in the Blk1 null space.
- bar 4 tells us that around 70% of the activity in the Blk2 row space comes from activity that used to be in the Blk1 null space.

**Comment by mobeets** [comment]
two examples of errors in YN2 in terms of what used to be in NB1 vs what used to be in RB1
(the third bar above).

so errors of YN2 that used to be in RB1 make up a smaller proportion of YN2, so they have lower error here, but also, it appears to follow the same overall pattern

![screen shot 2016-05-06 at 10 36 33 am](https://cloud.githubusercontent.com/assets/1677179/15076194/71fc1d4e-1376-11e6-9a73-b7b279b985cc.png)

**Comment by mobeets** [comment]
and now looking to see if different hypotheses explain one of these better than the other, it doesn't look like it:

![screen shot 2016-05-06 at 10 43 57 am](https://cloud.githubusercontent.com/assets/1677179/15076395/77d6f2e2-1377-11e6-80b8-7f9ace84c867.png)

(n.b. YN2 in NB1 looks about the same as the normal YN2 error, so I don't show it here)

**Comment by mobeets** [comment]
Other dates (note that 20120601 appears to be an exception to the rule that the same hypotheses work in both cases):

![screen shot 2016-05-06 at 10 48 22 am](https://cloud.githubusercontent.com/assets/1677179/15076560/54c40a46-1378-11e6-8fef-3fd26bbff38b.png)
![screen shot 2016-05-06 at 10 47 23 am](https://cloud.githubusercontent.com/assets/1677179/15076559/54c3ad76-1378-11e6-870c-c5a82ea53653.png)
![screen shot 2016-05-06 at 10 46 34 am](https://cloud.githubusercontent.com/assets/1677179/15076565/5a7d3c8c-1378-11e6-91d0-74987715dc10.png)
![screen shot 2016-05-06 at 10 46 12 am](https://cloud.githubusercontent.com/assets/1677179/15076566/5a7efb1c-1378-11e6-8cc2-a15bdb97e5f2.png)

**Comment by mobeets** [comment]
Now, doing all fits where for both Blk1 and Blk2 latents I either consider only the part in NB1, or only the part in RB1. Results:

0525: NB1 pruning best, RB1 pruning perfect
0601: NB1 pruning best, RB1 a mess
0709: NB1 pruning/hab best, RB1 pruning/cloud best
1125: NB1 pruning best, RB1 pruning/cloud best
1205: NB1 pruning best, RB1 pruning perfect

**Comment by mobeets** [comment]
so the "in NB1" version, for example, is about fitting the null space of Blk2 (`Y2*NB2`) using only the activity in Blk1 that was in NB1 (`Y1*(NB1*NB1')`). The "in RB1" version is using activity that used to move the cursor in Blk1 to predict activity that now doesn't move the cursor in Blk2.

![screen shot 2016-05-06 at 12 36 35 pm](https://cloud.githubusercontent.com/assets/1677179/15079887/2c20fa7a-1389-11e6-8f4f-be43f13c5daa.png)
![screen shot 2016-05-06 at 12 39 48 pm](https://cloud.githubusercontent.com/assets/1677179/15079888/2c22553c-1389-11e6-853f-60ca132c1ddb.png)
![screen shot 2016-05-06 at 12 36 48 pm](https://cloud.githubusercontent.com/assets/1677179/15079889/2c25e5b2-1389-11e6-880b-1785b4d9c7da.png)
![screen shot 2016-05-06 at 12 37 47 pm](https://cloud.githubusercontent.com/assets/1677179/15079890/2c264778-1389-11e6-9cf3-f318517b5f52.png)
![screen shot 2016-05-06 at 12 39 29 pm](https://cloud.githubusercontent.com/assets/1677179/15079891/2c2a1628-1389-11e6-9fd0-9d5d8ddb5f58.png)

**Comment by mobeets** [comment]
as a way of distinguishing these two attempts at separating activity in NB1 vs parts in RB1:
- first method fits normally, but evaluates errors in terms of the parts in NB1 vs parts in RB1
- second method (directly above) fits and evaluates in those terms

**Comment by mobeets** [comment]
but pruning appears to be the best in all these cases...
